haas/ext/obm/ipmi.py

```diff
@@ -86,10 +88,13 @@ def power_off(self):
         if self._ipmitool(['chassis', 'power', 'off']) != 0:
             raise OBMError('Could not power off node %s', self.label)
 
+    def require_legal_bootdev(self, dev):
+        if dev not in self.valid_bootdevices:
+            raise BadArgumentError('Invald boot device')
+
     @no_dry_run
     def set_bootdev(self, dev):
-        valid_bootdevices = ['disk', 'pxe', 'none']
-        if dev not in valid_bootdevices:
+        if dev not in self.valid_bootdevices:
             raise BadArgumentError('Invald boot device')
```

Comment:
I might just call require_legal_bootdev here, to make it explicit that this is the same thing. Might otherwise be a bit confusing to the reader whether this is meant to check something else.